README.md

```diff
@@ -1,2 +1,63 @@
-# fxa-geodb
-Firefox Accounts IP geolocation library
+## Firefox Accounts GeoDB Repo for Geolocation based services
+Provides a wrapper around [node-maxmind] (https://github.com/runk/node-maxmind) for getting the location data in an easy-to-read format.
+
+### Getting started
+Clone/fork the repo and run `npm i`. Then, include the module in your source file, like so:
+
+```JavaScript
+var GeoDB = require('./fxa-geodb');
+```
+--
+### API
+The function returns a promise that may either resolve (on successful finding of location data) or reject (if either the ip was invalid, or location data could not be found). Call the function, like so:
+
+```JavaScript
+GeoDB(ip).then(function (location) {
+		// success, resolved
+		// location data is available here
+      }, function (err) {
+      	// rejected :(
+      	// Uh-oh error
+      });
+```
+
+If you want to pass a custom timestamp to the function, do like so:
+```JavaScript
+GeoDB(ip, timestamp).then(function (location) {
+		// success, resolved
+		// location data is available here
+      }, function (err) {
+      	// Uh-oh error
+      });
+```
+where `timestamp` can be any JavaScript compatible representation of time.
```

Comment:
`timestamp` is given so that we can send an email with custom timestamp, say, after a week or so. Was suggested by @rfk 
Sure!, I will update with an example of a timestamp.